File: Arrays/Algorithms/Rearrangement/partition_array.cpp
#include <bits/stdc++.h>
using namespace std;

void partitionArray(vector<int>& arr, int x) {
	int i = 0;
	for (int j = 0; j < arr.size(); j++) {
		if (arr[j] < x) {
			swap(arr[i++], arr[j]);
		}
	}
}

int main() {
	int n, x;
	cin >> n >> x;
	vector<int> arr(n);
	for (int i = 0; i < n; i++) cin >> arr[i];
	partitionArray(arr, x);
	for (int v : arr) cout << v << ' ';
	cout << endl;
	return 0;
}

File: Arrays/Algorithms/Rearrangement/partition_array.py
def partition_array(arr, x):
	i = 0
	for j in range(len(arr)):
		if arr[j] < x:
			arr[i], arr[j] = arr[j], arr[i]
			i += 1

if __name__ == "__main__":
	n, x = map(int, input().split())
	arr = list(map(int, input().split()))
	partition_array(arr, x)
	print(*arr)
